sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/impl/HibernateRunDao.java

```diff
@@ -0,0 +1,386 @@
+  @Override
+  public long save(Run run) throws IOException {
+    updateWatchers(run);
+    long id;
+    if (run.getId() == AbstractRun.UNSAVED_ID) {
+      currentSession().save(run);
+      try {
+        String name = namingScheme.generateNameFor(run);
+        run.setName(name);
+
+        DbUtils.validateNameOrThrow(run, namingScheme);
+        Number newId = (long) currentSession().save(run);
+        if (newId.longValue() != run.getId()) {
+          log.error("Expected Run ID doesn't match returned value from database insert: rolling back...");
+          currentSession().delete(run);
+          throw new IOException("Something bad happened. Expected Run ID doesn't match returned value from DB insert");
+        } else {
+          id = run.getId();
+        }
+      } catch (MisoNamingException e) {
+        throw new IOException("Cannot save Run - issue with generating name");
+      }
+    } else {
+      currentSession().update(run);
+      id = run.getId();
+    }
+    return id;
+  }
+
+  @Override
+  public Run get(long id) throws IOException {
+    return (Run) currentSession().get(RunImpl.class, id);
+  }
+
+  @Override
+  public Run lazyGet(long id) throws IOException {
+    return get(id);
+  }
+
+  @Override
+  public List<Run> listAll() throws IOException {
+    Criteria criteria = currentSession().createCriteria(RunImpl.class);
+    @SuppressWarnings("unchecked")
+    List<Run> records = criteria.list();
+    return records;
+  }
+
+  @Override
+  public int count() throws IOException {
+    Criteria criteria = currentSession().createCriteria(RunImpl.class);
+    return ((Long) criteria.setProjection(Projections.rowCount()).uniqueResult()).intValue();
+  }
+
+  @Override
+  public boolean remove(Run run) throws IOException {
+    if (run.isDeletable()) {
+      currentSession().delete(run);
+
+      Run testIfExists = get(run.getId());
+      return testIfExists == null;
+    } else {
+      return false;
+    }
+  }
+
+  @Override
+  public Run getLatestStartDateRunBySequencerPartitionContainerId(long containerId) throws IOException {
+    Criteria criteria = currentSession().createCriteria(RunImpl.class);
+    criteria.createAlias("containers", "spc");
+    criteria.add(Restrictions.eq("spc.id", containerId));
+    criteria.addOrder(Order.desc("status.startDate"));
+    criteria.setMaxResults(1);
+    return (Run) criteria.uniqueResult();
+  }
+
+  @Override
+  public Run getLatestRunIdRunBySequencerPartitionContainerId(long containerId) throws IOException {
+    Criteria criteria = currentSession().createCriteria(RunImpl.class);
+    criteria.createAlias("containers", "spc");
+    criteria.add(Restrictions.eq("spc.id", containerId));
+    criteria.addOrder(Order.desc("run.id"));
+    criteria.setMaxResults(1);
+    return (Run) criteria.uniqueResult();
+  }
+
+  private static final String[] searchProperties = new String[] { "name", "alias", "description" };
+
+  private Criterion searchRestrictions(String querystr) {
```

Comment:
You can use `criteria.add(DbUtils.searchRestrctions(querystr, "name", "alias", "description")`.